### PR TITLE
Loki: Add support for "label_values(log stream selector, label)" in templating

### LINKED ISSUE #25205: Loki: label_values() - Use series API instead of labels API
**What would you like to be added**:

Ability to filter variables in loki using other variables. 

Ie in prometheus i can use 
```
label_values(disk_used_percent{host=~"$server"},path)
```
Where $server is
```
label_values(cpu_usage_idle, host)
```
To get list of mountpoints on this specific server. 

Will be nice to have same feature for loki, for example i have promtail collecting systemd journal on all machines, and needless to say they have different services running. Using label_values(unit) lists tons of services and it's not easy to understand how they correlate to specific servers.

Looks like loki already have matching functionality built-in, providing only units running on that node in case i already specified host:

image

**Why is this needed**:
Much cleaner selectors in dashboards and overall usability improvement

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
This PR adds support for `label_values(log stream selector, label)` pattern in templating. We are using [/series endpoint](https://grafana.com/docs/loki/latest/api/#series) for this and it will allow users to do following templating queries:

- `label_values({app="foo", namespace="prod"},instance)`
- `label_values({compose_service=~$service, compose_project=~$project}, container_name)`  - nested variables

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/25205

**Special notes for your reviewer**:
How to test this:
1. Go to Dashbboard -> Create variables
2. Choose Loki and try following queries (depending on your instance):
`label_values(job)`
`label_values({job="grafana"}, instance)`

### CODE REVIEW

File: public/app/plugins/datasource/loki/datasource.test.ts [hunk 1]
```diff
@@ -629,13 +633,3 @@ function makeAnnotationQueryRequest(options: any): AnnotationQueryRequest<LokiQu
     rangeRaw: timeRange,
   };
 }
-
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/datasource.test.ts:632]: I have removed this as we removed running of different queries for Loki v0 and Loki  v1. It was removed while ago and therefore these tests are not necessary anymore.

**Review by achatterjee-grafana — APPROVED** [review]
Doc content looks good.

**Review by ifrost — APPROVED** [review]
b

**Comment by JessedeJonge** [comment]
Which version of Grafana has this change?

**Comment by ivanahuckova** [comment]
This is currently merged in master and will be part of Grafana 8.1 release. You can always check it out by looking at assigned Milestone.

**Comment by JessedeJonge** [comment]
@ivanahuckova Thanks, my bad. Looking forward to it!

**Comment by alanprot** [comment]
Is there a reason to use `/series` instead os `label/<label_name>/values` here?

Both of the apis support filtering by `match[]=<series_selector>` and if we use the latter the response will be way smaller.

**Pull request merged**